src/cmake/properties.cmake

```diff
@@ -0,0 +1,50 @@
+## Based on https://github.com/GameFoundry/bsf/blob/master/Source/CMake/Properties.cmake
+## which is covered under the MIT License, same as GGPO.
+
+include(CheckCXXCompilerFlag)
+
+# Configuration types
+if(NOT CMAKE_CONFIGURATION_TYPES) # Multiconfig generator?
+	if(NOT CMAKE_BUILD_TYPE)
+		message(STATUS "Defaulting to release build.")
+		set_property(CACHE CMAKE_BUILD_TYPE PROPERTY VALUE "Release")
+	endif()
+endif()
+
+if(${CMAKE_SIZEOF_VOID_P} EQUAL 8)
+	set(GGPO_64BIT true)
+endif()
+
+## GGPO uses C-with-classes C++ flavour, so technically it should compile with C++03
+## but to be more safe, we set the standard to C++11
+set(CMAKE_CXX_STANDARD 11)
```

Comment:
I see no problem with modernising the codebase somewhat (and bringing up the cpp standard). However, I know Tony has expressed concerns with bringing in the STL so this is a discussion that we should probably have at some point.

Reply:
Can understand. STL while convenient, does add some bloat. I just set it to C++11 to be more on the safe side, but I will try to lower it down to C++03.

Reply:
Changed the set C++ standard to C++98 (CMake doesn't define C++03, but GGPO compiles fine with C++98).